tasks/main.yml

```diff
@@ -16,9 +16,8 @@
       ansible_ssh_user: "{{ bootstrap_user }}"
     check_mode: no
     when:
-      - ansible_fqdn is undefined # Test if fact checking has happened already
-      - login_as_self.rc is defined
-      - login_as_self.rc != 0
+      - login_as_self is defined
+      - login_as_self | failed
```

Comment:
This breaks my test [1]. Can we keep this as login_as_self.rc != 0 or why is this changed?

[1]: https://gist.github.com/martbhell/da166decadc5c8c13d2b5ddb0c0d098f
It's a copy of ansible-role-users/tests/test.yml with some changes to disable facts gather and set bootstrap_user. I have also removed any settings that apply from my ~/.ssh/config. 

This test playbook fails (can't ssh into instance) with this PR's changes because the role doesn't change ansible_ssh_user to {{ bootstrap_user }} even though it can't ssh as my own user into the instance.

Reply:
The chnage was aftera discussio nwith Kalle, the 2 conditionla we had here orginally never wokred as per my commit message and he suggested it might be better to set this conditional as:

If login as self is defined and login failed as I understand that setting to be  mean, then set the bootstrap_user. This has been working in rebuild of the rally server when the roles runs for the fits time it sets the user to rot. If you think this logic doesn't work then I can change it back.

For Ansible ver 2.5 the new way to use a `<value | failed>` is `<value is failed>`. This may read better  itnhe code, what do you think?

Reply:
The previous task only fails in certain conditions and we don't ignore_errors in it.
If the task with register: login_as_self has failed: True then that task has failed the play and we would never end up at the second task.

Reply:
ok I will change that back.